### DISCUSSION TITLE
[Select] How to display different value in labels and in selected input?

### DISCUSSION THREAD

**Discussion by bombillazo** [ant-design/ant-design#41691] [General]
Hello, I am trying to achieve the following:

The select component shows an abbreviated value:
<img width="458" alt="image" src="https://user-images.githubusercontent.com/44002321/230500517-7852e02b-bc33-4e82-b86c-5c6cd5bb6b18.png">

When the dropdown is open, the labels show the full names:
<img width="449" alt="image" src="https://user-images.githubusercontent.com/44002321/230502042-c711c5a3-5cc5-4da3-9590-7d1769725001.png">

At the moment, the label is shown in the Select when it is closed; how can I modify or show another value based on the selected value when the Select dropdown is closed?

**Comment by bombillazo** [comment]
Is this something possible with the Select component and props?

**Comment by linxianxi** [comment]
try https://codesandbox.io/s/ding-zhi-hui-tian-nei-rong-antd-5-6-0-forked-cdejo4

**Reply by linxianxi** [reply]
if you want to use virtual scroll, you can set number
<img width="1245" alt="image" src="https://github.com/ant-design/ant-design/assets/47104575/c696631c-5ad1-41c8-9d79-04bb864b54d5">

**Reply by bombillazo** [reply]
Thanks, that kind of works, but now I cannot filter by the full name.

**Reply by bombillazo** [reply]
Nevermind! I got it to work by using:
```ts
optionFilterProp="children"
optionLabelProp="label"
```

Thank you!

**Comment by neerajrathore** [comment]
Hi @bombillazo hope you are doing well and thanks for your solution.

> Nevermind! I got it to work by using:
> 
> ```ts
> optionFilterProp="children"
> optionLabelProp="label"
> ```
> 
> Thank you!

 I am having trouble in creating this kind of input filed that also have country dropdown select and input number field. Can you suggest the way how you created it?

> <img alt="image" width="449" src="https://user-images.githubusercontent.com/44002321/230502042-c711c5a3-5cc5-4da3-9590-7d1769725001.png">

**Comment by duchicong** [comment]
Default select alway `children`. But, If you want show `value` when selected then you can use this:

`optionLabelProp="value"`
